texparser/tar_extractor.py

```diff
@@ -55,11 +63,21 @@ def untar_file_into_folder(self, file_name: str):
         return self.extract_file_into_folder(file_name, "xvf", ".tar")
 
     def create_extract_folder_path(self) -> None:
-        if not os.path.exists(self.extract_folder_path) and self.extract_folder_path != "":
-            os.mkdir(self.extract_folder_path)
-        if not os.path.exists(self.extract_folder_path + self.dataset_folder_path) and self.dataset_folder_path != "":
-            os.mkdir(self.extract_folder_path + self.dataset_folder_path)
+        try:
+            if not os.path.exists(self.extract_folder_path) and self.extract_folder_path != "":
+                os.mkdir(self.extract_folder_path)
+            if not os.path.exists(self.extract_folder_path + self.dataset_folder_path) and self.dataset_folder_path != "":
```

Comment:
try to avoid stuff like that and use os.path.join otherwise the code will fail on windows because of the \ in the filepaths vs /
https://www.geeksforgeeks.org/python-os-path-join-method/
plus dont compute twice make a variable first then reference to it instead of creating the path on the fly twice